### DISCUSSION TITLE
Let's Talk Launchers

### DISCUSSION THREAD

**Discussion by Archfx** [Archfx/duo-de#56] [Show and tell]
With the duo-de, I kept the Trebuchet launcher due to the AOSP look and feel. But there may be certain scenarios that the Trbuchet launcher is not the solution.

So I am creating this thread to share your opinions and experiences about any third-party launchers that you have tried on Surface Duo (particularly with this ROM).

**Comment by Archfx** [comment]
These are two launchers that motivated the current state of the duo-de

- [Taskbar](https://github.com/farmerbb/Taskbar)
- [Smartdock](https://github.com/axel358/smartdock)

Both are really great projects, however, Taskbar has not been in active development lately. 

This is a screenshot of an earlier version of this ROM with Smartdock. It does have pretty nice features.
![src-duo](https://github.com/user-attachments/assets/71d61444-3c39-4e73-9fa8-caa80563d731)

**Reply by Ethanol10** [reply]
Smartdock does look a lot more polished over taskbar

**Reply by Archfx** [reply]
Yep, I think the taskbar was not maintained after a certain point making it a bit outdated.

**Reply by Ethanol10** [reply]
I did try out smartdock, but I couldn't consistently get it to activate as the underlying gesture for multitasking was being triggered.

**Reply by Archfx** [reply]
I think it has multiple ways to launch applications, a hover S button, and both from the taskbar. One thing I noted is that it needs to draw on top of other applications, android doesn't like that and complains a lot.

**Reply by GumbyXGames** [reply]
> Yep, I think the taskbar was not maintained after a certain point making it a bit outdated.

It was updated 3 week ago...
https://github.com/farmerbb/Taskbar/releases/tag/207

**Comment by GumbyXGames** [comment]
Is possible to sideload Pixel Launcher to possibility get single-screen apps working like on stock A12?

**Comment by sir86** [comment]
on stock A12 its Microsoft Launcher but if you install this launcher from PS on A14 or A15 DUO its crash on lunch . Looks like MS made custom Launcher version just for DUO1/DUO2 and the version from PS not work on any foldable phones.

MS made it actually very good on DUO phones. There is no single-screen apps that works on any other foldable phone like work on DUO. Samsung or Google still makes it works worse than on DUO.

its cool to have A14/15 on DUO but in other hand snigle screen apps feature from stock DUO firmware is very missing.

**Reply by radamofficial** [reply]
Do you think there is a way to return that functionality to the custom ROM? I'd love to learn and help with that

**Reply by Archfx** [reply]
Yes, it is possible. Display is handled by this [patch](https://github.com/Archfx/duo-de/blob/main-15/patches/duo/platform_frameworks_base/0002-ARCHFX-DUO-DE-posture-processor-integration.patch). Here it treats the two panels as a one massive display. But Android supports multiple displays, which we can treat as two panels separately. So the implementation should be changed and tested.

**Reply by radamofficial** [reply]
I'll look into it. But, is it ready to be applied to the ROM even with bugs?

**Comment by dmitrc** [comment]
@Archfx, is there a way to increase the size of the default freeform windows in the built-in launcher at runtime, or does that require altering the patch and rebuilding?
